File: src/main/scala/day02/source/Flink04_Source_Kafka.scala
package day02.source

import org.apache.flink.api.common.serialization.SimpleStringSchema
import org.apache.flink.streaming.api.datastream.DataStreamSource
import org.apache.flink.streaming.api.environment.StreamExecutionEnvironment
import org.apache.flink.streaming.connectors.kafka.FlinkKafkaConsumer
import org.apache.kafka.clients.consumer.ConsumerConfig

import java.util.Properties

object Flink04_Source_Kafka {
  def main(args: Array[String]): Unit = {
//    val env: StreamExecutionEnvironment = StreamExecutionEnvironment
//    val properties: Properties = new Properties()
//    properties.setProperty(ConsumerConfig.BOOTSTRAP_SERVERS_CONFIG,"hadoop102:9092")
//    properties.setProperty(ConsumerConfig.GROUP_ID_CONFIG,"flink_test")

//    val kafkaStream: DataStreamSource[String] = env.addSource(new FlinkKafkaConsumer("sensor", new SimpleStringSchema(), properties))
//    kafkaStream.map(record => {
//      record.
//    })
  }
}

File: src/main/java/com/atguigu/day02/source/Flink04_Source_Kafka.java
package com.atguigu.day02.source;

import org.apache.flink.api.common.functions.MapFunction;
import org.apache.flink.api.common.serialization.SimpleStringSchema;
import org.apache.flink.configuration.Configuration;
import org.apache.flink.streaming.api.datastream.DataStreamSource;
import org.apache.flink.streaming.api.datastream.SingleOutputStreamOperator;
import org.apache.flink.streaming.api.environment.StreamExecutionEnvironment;
import org.apache.flink.streaming.connectors.kafka.FlinkKafkaConsumer;
import org.apache.kafka.clients.consumer.ConsumerConfig;

import java.util.Properties;

public class Flink04_Source_Kafka {
    public static void main(String[] args) throws Exception {
        // 获取执行环境
//        StreamExecutionEnvironment env = StreamExecutionEnvironment.getExecutionEnvironment();
        StreamExecutionEnvironment env = StreamExecutionEnvironment.createLocalEnvironmentWithWebUI(new Configuration());

        // 设置并行度为1
        env.setParallelism(2);

        Properties properties = new Properties();
        properties.put(ConsumerConfig.GROUP_ID_CONFIG,"flink-test");
        properties.put(ConsumerConfig.BOOTSTRAP_SERVERS_CONFIG,"hadoop102:9092");

        // 添加source源
        DataStreamSource<String> source = env.addSource(new FlinkKafkaConsumer<String>("sensor", new SimpleStringSchema(), properties)).setParallelism(2);

        SingleOutputStreamOperator<WaterSensor> map = source.map(new MapFunction<String, WaterSensor>() {
            @Override
            public WaterSensor map(String value) throws Exception {
                String[] split = value.split(" ");
                return new WaterSensor(split[0], Long.parseLong(split[1]), Integer.parseInt(split[2]));
            }
        });

        map.print();

        env.execute();

    }
}
